plugins/modules/kafka_connectors.py

```diff
@@ -124,30 +126,66 @@ def get_connector_status(connect_url, connector_name, timeout):
     time.sleep(WAIT_TIME_BEFORE_GET_STATUS)
     status_url = "{}/{}/status".format(connect_url, connector_name)
 
-    res = open_url(status_url, validate_certs=False, timeout=timeout)
-    current_status = json.loads(res.read())
+    time_waited = 0
+    try:
+        connector_status = 'awaiting'
+        while connector_status != RUNNING_STATE and time_waited < TIMEOUT_WAITING_FOR_TASK_STATUS:
+            # retry until the connector is running or timeout
+            res = open_url(status_url, validate_certs=False, timeout=timeout)
+            current_status = json.loads(res.read())
 
-    connector_status = current_status['connector']['state']
+            connector_status = current_status['connector']['state']
 
-    failures = []
-    if connector_status != RUNNING_STATE:
-        failures.append("connector state paused or failed")
+            if connector_status != RUNNING_STATE:
+                time_waited += 1
+                time.sleep(1)
 
-    nb_tasks = len(current_status['tasks'])
-    time_waited = 0
+        if connector_status != RUNNING_STATE:
+            return False, "Connector state paused or failed"
+
+        nb_tasks = len(current_status['tasks'])
+
+    except urllib_error.HTTPError as e:
```

Comment:
The code for exception handling looks same in both the changes. Lets introduce a method for exception handling.